### PR TITLE
[FLINK-13430][build] Configure sending travis build notifications to builds@flink.apache.org

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

Send travis build notifications the builds@flink.apache.org mailing list. 

## Brief change log

Integrate builds mailing list to the travis yaml according to the [travis doc](https://docs.travis-ci.com/user/notifications/#configuring-email-notifications).

## Verifying this change

This change is trivial without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
I setup a CRON job on my testing branch to test the email setting: https://travis-ci.org/wuchong/flink/builds/563609665

**Comment by zentol** [comment]
how can we prevent this from sending notifications for contributor repositories?

**Comment by wuchong** [comment]
That's a good question @zentol . How do we handle this for slack channel?

**Comment by beyond1920** [comment]
@zentol @wuchong Sending notifications for contributor repositories as well seems not be useless as it looks like. 
I could receives build notifications from my own Flink repository and public Flink repository by 
subscribe builds mail-list and custom my own email forwarding rules.
Otherwise, I may need custom travis config in my own repository.

**Comment by zentol** [comment]
@wuchong the slack channel credentials are bound to the repository, and are simply invalid in all other repositories.

Having all contributor builds in builds@flink.apache.org just turns it into a massive spam fest, with the majority of builds not being relevant to the majority of subscribers.

**Comment by zentol** [comment]
Could you try this instead and check whether there still are notifications for contributor forks?
```
  email:
    recipients:
        secure: Tnf61eG2XC4uH9VOS2FnmDZVKMQqoXxQh1tPBuj7wcXCxSBtRKZtwrLHeQYDaH+cdDvy4rgzInKxq0imUyVnZ3vl3eTJyCtUW33RanUN8g4zZANy1nj9NbyhE7U8MCkIOMEuIkqGrI1XfFU0k+OMKaeLlUWtNUkBxeZVueL0iP0=
    on_success: change
    on_pull_requests: false
```

**Comment by wuchong** [comment]
Thanks @zentol , I will try that.

**Comment by wuchong** [comment]
> Could you try this instead and check whether there still are notifications for contributor forks?
> (and check whether it still works for apache/flink by pushing this PR into a temporary branch)
> 
> ```
>   email:
>     recipients:
>         secure: Tnf61eG2XC4uH9VOS2FnmDZVKMQqoXxQh1tPBuj7wcXCxSBtRKZtwrLHeQYDaH+cdDvy4rgzInKxq0imUyVnZ3vl3eTJyCtUW33RanUN8g4zZANy1nj9NbyhE7U8MCkIOMEuIkqGrI1XfFU0k+OMKaeLlUWtNUkBxeZVueL0iP0=
>     on_success: change
>     on_pull_requests: false
> ```

Good news! The secure works. The fail notification from my branch wuchong/flink:builds is sent to my own mail address.  And the notification from apache/flink:builds-test is sent to builds@flink.apache.org.

**Comment by zentol** [comment]
Well that sounds perfect then, +1.

**Comment by wuchong** [comment]
> Well that sounds perfect then, +1.

However, we still need to figure out how to accept builds emails. Currently, all notification from builds@travis-ci.org are rejected.

**Comment by zentol** [comment]
@wuchong see https://apache.org/dev/committers.html#mail-moderate

**Comment by wuchong** [comment]
Yes @zentol . I have read it and tried to reply **-accept** and **-allow** address. But it still doesn't work.

**Comment by wuchong** [comment]
I checked the builds@travis-ci.org has already been in the subscriber list and allow list.

**Comment by zentol** [comment]
@wuchong Could you reach out to INFRA on what the problem might be? As far as I can tell we should've setup everything correctly.

**Comment by wuchong** [comment]
Thanks for the advice @zentol , I have created an INFRA issue INFRA-18843.

**Comment by wuchong** [comment]
Hi @zentol , the INFRA guys have figured out the problem and the builds@travis-ci.org works now. I removed the fast-fail commit. 

I think it's ready to be merged now. 

Cheers,
Jark

**Comment by zentol** [comment]
+1 from my side.

**Comment by wuchong** [comment]
I would like to hold this PR a bit moment @zentol . Because I find it will lose the notification to my own email address for the branches in my own repo. 

After reading the Travis documentation, I think we will also lose the build notification to the committer who pushed the commit, which I think is important.
image

**Comment by zentol** [comment]
I had hoped this wouldn't apply to contributor builds since the mail was encrypted...

oh well, back to the drawing board instead. I guess we'll have to setup some kind of proxying over slack or similar for this to work then.

**Comment by wuchong** [comment]
Hi @zentol , I have setup a webhook to receive travis' notifications and generate a email and send to mailing list. 

All the things looks good (send to my own email address), I will test on the mailing list then.

File: flink-core/src/main/java/org/apache/flink/api/common/accumulators/Accumulator.java [hunk 1]
```diff
@@ -1,20 +1,4 @@
-/*
```
Comment (TisonKun) [inline flink-core/src/main/java/org/apache/flink/api/common/accumulators/Accumulator.java]: Accidentally removal I think. Revert to pass the checker :-)

Comment (wuchong) [reply flink-core/src/main/java/org/apache/flink/api/common/accumulators/Accumulator.java]: Thanks. I will revert this before merge. This is used to fail the travis fast.

Comment (TisonKun) [reply flink-core/src/main/java/org/apache/flink/api/common/accumulators/Accumulator.java]: Aha, I think we will back to the thread reduce building time :-)

**Comment by zentol** [comment]
@wuchong awesome, looking forward to seeing this in action. Once the mail passes through moderation and retains the formatting when posted on builds@flink.apache.org, feel free to merge this.

Can you expand on how you setup the webhook? Just to ensure we can set this up elsewhere if anything happens to your setup.

**Comment by wuchong** [comment]
@zentol , Sure, here is the source code: https://github.com/wuchong/flink-notification-bot
I deployed it in my VM.

**Comment by wuchong** [comment]
Once it is done. I can contribute it to, e.g., https://github.com/flinkbot

**Comment by wuchong** [comment]
Progress Update: 

The mailing list doesn't accept "text/html" context type. So I send a request to the INFRA to enable html the builds.  [NFRA-18918](https://issues.apache.org/jira/browse/INFRA-18918)

**Comment by wuchong** [comment]
The HTML is enabled. And I tested several emails and looks good now. I will merge this then.

**Pull request closed without merging**